chain/chain/src/trie_store_resharder.rs

```diff
@@ -0,0 +1,164 @@
+//! Logic for writing trie nodes with the appropriate shard prefix.
+//! This logic runs in parallel to chain processing.
+
+use crate::types::RuntimeAdapter;
+use near_chain_configs::{MutableConfigValue, ReshardingConfig};
+use near_primitives::{hash::CryptoHash, state};
+use near_store::{ShardUId, adapter::StoreAdapter, flat::ParentSplitParameters};
+use std::sync::Arc;
+
+pub struct TrieStoreResharder {
+    runtime: Arc<dyn RuntimeAdapter>,
+    /// Configuration for resharding.
+    resharding_config: MutableConfigValue<ReshardingConfig>,
+}
+
+impl TrieStoreResharder {
+    pub fn new(
+        runtime: Arc<dyn RuntimeAdapter>,
+        resharding_config: MutableConfigValue<ReshardingConfig>,
+    ) -> Self {
+        Self { runtime, resharding_config }
+    }
+
+    fn split_shard_task_impl(
+        &self,
+        shard_uid: ShardUId,
+        state_root: CryptoHash, // TODO: fix
+        parent_split_parameters: ParentSplitParameters,
+    ) -> Result<(), String> {
+        // Determines after how many bytes worth of key-values the process stops to commit changes
+        // and to check cancellation.
+        let batch_size = self.resharding_config.get().batch_size.as_u64() as usize;
+        // Delay between every batch.
+        let batch_delay = self.resharding_config.get().batch_delay.unsigned_abs();
```

Comment:
heh I see the same in flat storage resharder, what's wrong with the properly typed ByteSize and Duration? ;) 

probably totally irrelevant at this stage of the PR